### DISCUSSION TITLE
Icon layer is not working anymore after new release of 8.4.2 version

### DISCUSSION THREAD

**Discussion by Mayoorshardha** [visgl/deck.gl#5452] [Q&A]
Hello Everyone!
I am using icon layer for my project and it was working fine up until 8.4.1 version. However, after the update to the deck gl to 8.4.2, the map is rendering however the icons are not showing anymore. I am getting an error of set uniforms is not defined.
I am getting data from an api route and iconmap and iconatlas are imported from my assets folder. Those are fine.

Package.json
"@deck.gl/core": "8.4.2",
 "deck.gl": "8.4.2",
  "@deck.gl/layers": "8.4.2",
  "@deck.gl/react": "8.4.2",
  "@luma.gl/core": "8.4.1",
  "@luma.gl/engine": "8.4.1",
  "@luma.gl/webgl": "8.4.1",

Code:
const MAP_VIEW = new MapView({ repeat: true });

const INITIAL_VIEW_STATE = {
  longitude: -95.5698,
  latitude: 29.7604,
  zoom: 8,
  maxZoom: 20,
  pitch: 0,
  bearing: 0,
};

 const mapStyle =
    "https://basemaps.cartocdn.com/gl/positron-gl-style/style.json";

 const [data, setData] = useState([]);
const layer = new IconLayer({
    id: "icon-layer",
    getIcon: (d) => "marker",
    sizeUnits: "meters",
    sizeScale: 2000,
    sizeMinPixels: 30,
    pickable: true,
    getPosition: d => d.coordinates,
    getSize: d => 5,
    data,
    iconAtlas,
    iconMapping
  });

  return (
    <DeckGL
      width={"100%"}
      layers={[layer]}
      views={MAP_VIEW}
      initialViewState={INITIAL_VIEW_STATE}
      controller={true}
      getTooltip={getTooltip}
      useDevicePixels={false}
    >
      <StaticMap
        reuseMaps
        mapStyle={mapStyle}
        mapboxApiAccessToken={TOKEN}
        preventStyleDiffing={true}
      />
    </DeckGL>
  );

**Comment by Pessimistress** [comment]
There was no change in 8.4.2 that affected the IconLayer. The website examples at https://deck.gl/examples/icon-layer/ and https://deck.gl/docs/api-reference/layers/icon-layer are using the latest release. I suggest you revert back to 8.4.1 and see if it indeed is caused by the upgrade of deck.gl.

**Reply by Mayoorshardha** [reply]
I tried 8.4.0 version but it did not work.
Deck gl keeps getting updated every week which will eventually lead to this problem in future too. I have locked the packages in locked-package.json and also tried to install a specific version but it does not work. Deck gl updated to 8.4.3.
Without doing npm install, deck gl updates itself

**Comment by Mayoorshardha** [comment]
With deck gl 8.4.3 version release, it got fixed on its own. Now its working fine!
